static/app/views/dashboardsV2/widgetBuilder/widgetBuilder.tsx

```diff
@@ -788,366 +764,83 @@ function WidgetBuilder({
+                  />
+                  {[DisplayType.TABLE, DisplayType.TOP_N].includes(state.displayType) && (
+                    <ColumnsStep
+                      dataSet={state.dataSet}
+                      queries={state.queries}
                       displayType={state.displayType}
-                      onChange={(option: {label: string; value: DisplayType}) => {
-                        handleDisplayTypeOrTitleChange('displayType', option.value);
-                      }}
-                      error={state.errors?.displayType}
-                    />
-                    <VisualizationWrapper displayType={state.displayType}>
-                      <WidgetCard
-                        organization={organization}
-                        selection={pageFilters}
-                        widget={currentWidget}
-                        isEditing={false}
-                        widgetLimitReached={false}
-                        renderErrorMessage={errorMessage =>
-                          typeof errorMessage === 'string' && (
-                            <PanelAlert type="error">{errorMessage}</PanelAlert>
-                          )
-                        }
-                        isSorting={false}
-                        currentWidgetDragging={false}
-                        noLazyLoad
-                      />
-                    </VisualizationWrapper>
-                  </BuildStep>
-                  <BuildStep
-                    title={t('Choose your data set')}
-                    description={t(
-                      'This reflects the type of information you want to use. For a full list, read the docs.'
-                    )}
-                  >
-                    <DataSetChoices
-                      label="dataSet"
-                      value={state.dataSet}
-                      choices={DATASET_CHOICES}
-                      disabledChoices={
-                        state.displayType !== DisplayType.TABLE
-                          ? [
-                              [
-                                DataSet.ISSUES,
-                                t(
-                                  'This data set is restricted to the table visualization.'
-                                ),
-                              ],
-                            ]
-                          : undefined
-                      }
-                      onChange={handleDataSetChange}
+                      organization={organization}
+                      widgetType={widgetType}
+                      queryErrors={state.errors?.queries}
+                      onQueryChange={handleQueryChange}
+                      onYAxisOrColumnFieldChange={handleYAxisOrColumnFieldChange}
+                      explodedFields={explodedFields}
+                      explodedColumns={explodedColumns}
+                      explodedAggregates={explodedAggregates}
+                      onGetAmendedFieldOptions={handleGetAmendedFieldOptions}
```

Comment:
I've moved it to the utils, but regarding the `withTags`, since two different components need it, IMO it's best if we keep it in the parent and just passes it down